core/util.go

```diff
@@ -217,61 +218,10 @@ func IsAnyNilOrZero(vals ...interface{}) bool {
 		case nil:
 			return true
 		// These are the go types which correspond to proto3's scalar fields.
-		case bool:
-			if v == false {
+		case bool, int, int32, int64, uint, uint32, uint64, float32, float64, string:
```

Comment:
One alternative here would be to move the type switch into its own function (which only checks one value at a time), so that it can be more streamlined:
```
for _, val := range vals {
    if isNilOrZero(val) {
        return true
    }
}

func isNilOrZero(val interface{}) bool {
    switch v := val.(type) {
        case nil: return true
    case int:
        return v == 0
    case int32:
        return v == 0
etc
```

This way the switch case would be compact (one line per case), without losing the early-return upon detection of the first nil or zero value in the inputs.